### PR TITLE
switch-from-non-containerized-to-containerized: mask unit files

### PR DESCRIPTION
**Pull request by leseb**
We must mask the image so we are sure that even if the system reboots
then the OSDs won't start.

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
@fultonj @gfidente

**Comment by gfidente** [comment]
I think we're hit by this on fresh deployments as well

File: roles/ceph-docker-common/tasks/pre_requisites/remove_redhat_ceph_packages.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+- name: list ceph redhat packages
+  shell: |
+    rpm -qa | grep -E "ceph|librbd|librados"
```
Comment (gfidente) [inline roles/ceph-docker-common/tasks/pre_requisites/remove_redhat_ceph_packages.yml]: seems too generic, we'd end up removing a lot of packages for cascading dependencies

we probably want to remove only the leaf packages here, ceph-mon, ceph-osd, -mds, -test, -radosgw, -mgr

File: roles/ceph-docker-common/tasks/pre_requisites/remove_ceph_packages.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
```
Comment (gfidente) [inline roles/ceph-docker-common/tasks/pre_requisites/remove_ceph_packages.yml]: maybe these should also be tagged with_pkg so the removal can be skipped

File: roles/ceph-docker-common/tasks/pre_requisites/remove_redhat_ceph_packages.yml (hunk 1 above)
Comment (ktdreyer) [reply roles/ceph-docker-common/tasks/pre_requisites/remove_redhat_ceph_packages.yml]: :+1: purging packages like this needs a lot of testing

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-update_cluster

**Comment by leseb** [comment]
@andrewschoen please review my new scenario addition.

**Comment by leseb** [comment]
jenkins test this please

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-switch_to_containers

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-switch_to_containers

File: tox.ini [hunk 3]
```diff
@@ -96,8 +96,24 @@ commands=
...
+      ceph_docker_image_tag={env:UPDATE_CEPH_DOCKER_IMAGE_TAG:latest} \
+      ceph_dev_branch={env:UPDATE_CEPH_DEV_BRANCH:master} \
+      ceph_dev_sha1={env:UPDATE_CEPH_DEV_SHA1:latest} \
+  "
+
+  bash -c 'export DOCKER=True && testinfra -n 4 --sudo -v --connection=ansible --ansible-inventory={changedir}/hosts {toxinidir}/tests/functional/tests'
+
 [testenv]
 whitelist_externals =
+    bash
```
Comment (andrewschoen) [inline tox.ini]: ``bash`` is already listed below

File: tox.ini [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [tox]
 envlist = {dev,jewel,luminous,rhcs}-{ansible2.2,ansible2.3}-{xenial_cluster,journal_collocation,centos7_cluster,dmcrypt_journal,dmcrypt_journal_collocation,docker_cluster,purge_cluster,purge_dmcrypt,docker_dedicated_journal,docker_dmcrypt_journal_collocation,update_dmcrypt,update_cluster,cluster,purge_docker_cluster,update_docker_cluster}
-  {dev,luminous}-{ansible2.2,ansible2.3}-{bluestore_journal_collocation,bluestore_cluster,bluestore_dmcrypt_journal,bluestore_dmcrypt_journal_collocation,bluestore_docker_cluster,bluestore_docker_dedicated_journal,bluestore_docker_dmcrypt_journal_collocation,lvm_osds,purge_lvm_osds,shrink_mon,shrink_osd}
+  {dev,luminous}-{ansible2.2,ansible2.3}-{bluestore_journal_collocation,bluestore_cluster,bluestore_dmcrypt_journal,bluestore_dmcrypt_journal_collocation,bluestore_docker_cluster,bluestore_docker_dedicated_journal,bluestore_docker_dmcrypt_journal_collocation,lvm_osds,purge_lvm_osds,shrink_mon,shrink_osd,switch_to_containers}
```
Comment (andrewschoen) [inline tox.ini]: if we moved ``switch_to_containers`` up a line we'd be able to run that scenario on ``rhcs`` too.

**Comment by andrewschoen** [comment]
jenkins test luminous-ansible2.3-switch_to_containers

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-switch_to_containers

**Pull request merged**